src/OpenTelemetry.Instrumentation.AWSLambda/Implementation/AWSLambdaUtils.cs

```diff
@@ -198,6 +193,11 @@ private static IEnumerable<string> GetHeaderValues(APIGatewayProxyRequest reques
         {
             return values;
         }
+        else if (request.Headers != null &&
```

Comment:
If we think we need to be case-insensitive, we need to be case-insensitive in the multi-header variant as well.

Reply:
Introduced a single method GetHeaderValues for each type of HTTP trigger and used in both places: as the propagator getter and here for header values extraction.